pkg/controller/build/build_controller.go

```diff
@@ -459,6 +471,20 @@ func (r *ReconcileBuild) validateBuildRunOwnerReferences(ctx context.Context, b
 	return nil
 }
 
+// validateSourceURL returns error message if remote repository doesn't exist
+func (r *ReconcileBuild) validateSourceURL(ctx context.Context, b *build.Build, sourceURL string) error {
+	switch b.GetAnnotations()[build.AnnotationBuildVerifyRepository] {
+	case "", "true":
+		return git.ValidateGitURLExists(sourceURL)
+	case "false":
+		ctxlog.Info(ctx, fmt.Sprintf("the annotation %s is set to %s, nothing to do", build.AnnotationBuildVerifyRepository, b.GetAnnotations()[build.AnnotationBuildVerifyRepository]))
+		return nil
+	default:
+		ctxlog.Error(ctx, fmt.Errorf("the annotation %s was not properly defined, supported values are true or false", build.AnnotationBuildVerifyRepository), namespace, b.Namespace, name, b.Name)
+		return fmt.Errorf("the annotation %s was not properly defined, supported values are true or false", build.AnnotationBuildVerifyRepository)
```

Comment:
Just to avoid that if we ever have to change the error message, that we have to do it in two spots.
```suggestion
		var annoErr = fmt.Errorf("the annotation %s was not properly defined, supported values are true or false", build.AnnotationBuildVerifyRepository)
		ctxlog.Error(ctx, annoErr, namespace, b.Namespace, name, b.Name)
		return annoErr
```